src/app/stack/card-details/card-details.component.ts

```diff
@@ -981,9 +1225,78 @@ export class CardDetailsComponent implements OnInit, OnChanges {
         return headers;
     }
 
+    // mComponentDetails->McomponentInformation commented
+    // private createDictionary(report: ResultInformationModel, whatCard: string): Object {
+    //     if (whatCard === 'security') {
+    //         let components: Array<ComponentInformationModel> = null;
+    //         if (report.user_stack_info
+    //             && report.user_stack_info.analyzed_dependencies
+    //             && report.user_stack_info.analyzed_dependencies.length > 0) {
+    //             components = report.user_stack_info.analyzed_dependencies;
+    //         }
+
+    //         const depDict = new Object();
+    //         const transitiveDeps = new Array();
+    //         components.forEach(dep => {
+    //             if (!dep.hasOwnProperty('transitive')) {
+    //                 let uidofDep = dep.name + "-" + dep.version;
+    //                 depDict[uidofDep] = null;
+    //             }
+    //         });
+
+    //         components.forEach(dep => {
+    //             if (dep.hasOwnProperty('transitive')) {
+    //                 transitiveDeps.push(dep);
+    //             }
+    //         });
+
+    //         transitiveDeps.forEach(dep => {
+
+    //             let tempDep = { ...dep };
+    //             tempDep.transitive = null;
+
+    //             if (dep.transitive && dep.transitive.isTransitive) {
+    //                 let affectedDirectDeps = dep.transitive.affected_direct_deps;
+
+    //                 affectedDirectDeps.forEach(dirDep => {
+    //                     let uidDirDep = dirDep.package + "-" + dirDep.version;
+
+    //                     if (depDict[uidDirDep] == null) {
+    //                         depDict[uidDirDep] = [new MComponentDetails(
+    //                             this.getComponentInformation(tempDep),
+    //                             null)]
+    //                     } else {
+    //                         depDict[uidDirDep].push(new MComponentDetails(
+    //                             this.getComponentInformation(tempDep),
+    //                             null));
+    //                     }
+
+    //                 });
+    //             }
+    //         });
+
+
+    //         const directDependenciesWithTransitveDetailsObj = new Object();
+
+    //         for (var key of Object.keys(depDict)) {
+    //             if (depDict[key] != null) {
+    //                 directDependenciesWithTransitveDetailsObj[key] = depDict[key];
```

Comment:
Can you remove all commented code?

Reply:
Removed most of the commented code. 
Few of the files still have commented code(test cases), kept the for reference.

Reply:
fixed